### ISSUE TITLE
Backoffice notifications not displaying

### ISSUE THREAD

**Issue by nathanwoulfe** [umbraco/Umbraco-CMS#10403]
### Which Umbraco version are you using?

V9 beta003

### Bug summary

I return a `List<BackOfficeNotification>` to display UI messages in response to editor actions. These are not displaying in v9 as the notification objects have a field named `NotificationType`, while the front-end notificationsService checks for `type`.

Not sure why this isn't working, given the BackOfficeNotification type hasn't changed for V9, nor has the AngularJs service.

### Steps to reproduce

Create a controller like so:

```csharp
public class NotificationsExampleController : UmbracoAuthorizedApiController {
  public IActionResult Get() {
    var simpleNotificationsModel = new SimpleNotificationsModel();
    simpleNotificationsModel.AddNotification("Header", "message", NotificationStyle.Info);

    return Ok(new
    {
        notifications = simpleNotificationsModel.Notifications
    });
  }
}
```

Modify umbraco.services.js, adding the below at line 3360 (quicker than adding a new package, all we need is the js to execute).

```javascript
  $http.get('/umbraco/backoffice/api/notificationsexample/get').then(resp => {
    resp.data.notifications.forEach(r => notificationsService.showNotification(r));
  });
```

### Expected result / actual result

I'd expect my notifications to render, instead I get console errors:

`args.type cannot be null`

This throws at line 4981 of umbraco.services.js:

![image](https://user-images.githubusercontent.com/3248070/120974993-1a0c8180-c7b4-11eb-83fe-f317da281d7b.png)

because the returned notifications look like so:

![image](https://user-images.githubusercontent.com/3248070/120975092-33153280-c7b4-11eb-9f90-e43f6b6567bb.png)

**Comment by rasmusjp** [comment]
This might be because the models are serialized using `System.Text.Json` which doesn't use the `DataContract` attributes

**Comment by nathanwoulfe** [comment]
That sounds likely, will have a proper look tomorrow. For now, I've extended the BackOfficeNotification object to add a Type field, which solves the problem but is a bit ugly.

**Comment by bergmania** [comment]
I have looked into this and you are right @rasmusjp.

 i wonder why all of our API controllers are not using the same serilization. Would it be a problem if all of them return the angular prefix and use the newtonsoft serialization.
 
 In that case this controller would return this:
```json
)]}',
{"notifications":[{"header":"Header","message":"message","type":"Info"}]}
```

**Comment by nikolajlauridsen** [comment]
Fixed in #11054
